app/controllers/users_controller.rb

```diff
@@ -1,5 +1,28 @@
 class UsersController < ApplicationController
 
+  def new
+    @user = User.new
+  end
+
   def show
   end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      session[:user_id] = @user.id
+      flash[:message] = "User has been successfully created!"
+      redirect_to user_path(@user)
+    else
+      flash[:errors] = @user.errors.full_messages(', ').join
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.